coa-helm/templates/cbas-api.yaml

```diff
@@ -28,22 +32,34 @@ data:
   {{ .Values.cbas.conf_file }}: |-
     spring:
       datasource:
+        {{- if .Values.postgres.podLocalDatabaseEnabled }}
         jdbc-url: jdbc:postgresql://{{ include "app.fullname" . }}-postgres:{{ .Values.postgres.port }}/{{ .Values.postgres.cbas.dbname }}
-        username: {{ .Values.postgres.user }}
         password: {{ include "dbPassword" . | b64enc | quote }}
+        {{- else }}
+        jdbc-url: "jdbc:postgresql://{{ .Values.postgres.host }}:{{ .Values.postgres.port }}/{{ .Values.postgres.cbas.dbname }}?sslMode=Require&authenticationPluginClassName=com.azure.identity.extensions.jdbc.postgresql.AzurePostgresqlAuthenticationPlugin"
```

Comment:
My DRY senses are tingling, especially since this is 99% coming from `values` anyway, but I'm not sure I see a neat way to do it without a new utility function, and I wouldn't suggest doing that for this PR